src/gt4py/next/program_processors/runners/dace_fieldview/transformations/gpu_utils.py

```diff
@@ -0,0 +1,374 @@
+        #   problem is, that Map fusion does not guarantee a certain order of Map
+        #   variables. Currently this is not a problem because of the way it is
+        #   implemented.
+        # TODO(phimuell): Fix the issue described above.
+        # TODO(phimuell): Maybe we should fuse trivial GPU maps before we do anything.
+        sdfg.apply_transformations_once_everywhere(
+            gtx_transformations.SerialMapPromoterGPU(),
+            validate=False,
+            validate_all=False,
+        )
+        sdfg.apply_transformations_repeated(
+            gtx_transformations.SerialMapFusion(
+                only_toplevel_maps=True,
+            ),
+            validate=validate,
+            validate_all=validate_all,
+        )
+
+    # Set the GPU block size if it is known.
+    if gpu_block_size is not None:
+        gt_set_gpu_blocksize(
+            sdfg=sdfg,
+            gpu_block_size=gpu_block_size,
+            gpu_launch_bounds=gpu_launch_bounds,
+            gpu_launch_factor=gpu_launch_factor,
+        )
+
+    return sdfg
+
+
+def gt_set_gpu_blocksize(
+    sdfg: dace.SDFG,
+    gpu_block_size: Optional[Sequence[int | str] | str],
+    gpu_launch_bounds: Optional[int | str] = None,
+    gpu_launch_factor: Optional[int] = None,
+) -> Any:
+    """Set the block sizes of GPU Maps.
+
+    Args:
+        sdfg: The SDFG to process.
+        gpu_block_size: The block size to use.
+        gpu_launch_bounds: The launch bounds to use.
+        gpu_launch_factor: The launch factor to use.
+    """
+    xform = GPUSetBlockSize(
+        block_size=gpu_block_size,
+        launch_bounds=gpu_launch_bounds,
+        launch_factor=gpu_launch_factor,
+    )
+    return sdfg.apply_transformations_once_everywhere([xform])
+
+
+def _gpu_block_parser(
+    self: "GPUSetBlockSize",
+    val: Any,
+) -> None:
+    """Used by the setter ob `GPUSetBlockSize.block_size`."""
+    org_val = val
+    if isinstance(val, tuple):
+        pass
+    elif isinstance(val, list):
+        val = tuple(val)
+    elif isinstance(val, str):
+        val = tuple(x.replace(" ", "") for x in val.split(","))
+    else:
+        raise TypeError(
+            f"Does not know how to transform '{type(val).__name__}' into a proper GPU block size."
+        )
+    if len(val) == 1:
+        val = (*val, 1, 1)
+    elif len(val) == 2:
+        val = (*val, 1)
+    elif len(val) != 3:
+        raise ValueError(f"Can not parse block size '{org_val}': wrong length")
+    try:
+        val = [int(x) for x in val]
+    except ValueError:
+        raise TypeError(
+            f"Currently only block sizes convertible to int are supported, you passed '{val}'."
+        ) from None
+    self._block_size = val
+
+
+def _gpu_block_getter(
+    self: "GPUSetBlockSize",
+) -> tuple[int, int, int]:
+    """Used as getter in the `GPUSetBlockSize.block_size` property."""
+    assert isinstance(self._block_size, (tuple, list)) and len(self._block_size) == 3
```

Comment:
Here I have a question. I have seen most fields are bi-dimensional. Why should we force a tri-dimensional block? Are we relying on the fact that the second and third dimensions are always the same?

Reply:
You are right most fields are, but GPU Kernels always have three dimensions, thus you need to return three values.

Reply:
I think the main reason why this is not an issue is that we assume there will be one single gpu map for the entire program. The time we start defining multiple cuda kernels within a program (by means of global temporaries), not necessarily all maps will have to use the same kernel block. Then, different definitions of 2d and 3d grids could be interesting.